### ISSUE TITLE
`rewrite` options should apply to all responses

### ISSUE THREAD

**Issue by tillsc** [nodejitsu/node-http-proxy#914]
According to the documentation the `hostRewrite` and `autoRewrite` options only apply to 3xx responses. But the `Location`-Header is allowed in more than only the 3xx responses (e.g. in `201 Created`).

I don't understand why it is necessary to limit this functionality to specific response codes. Why couldn't just every `Location` header be analyzed and replaced if there is one present in the response?

**Comment by gabrielboucher** [comment]
I agree, this is currently biting me. We are implementing an API that tries hard to follow this specification: http://jsonapi.org/format/#crud-creating-responses especially where it says "The response SHOULD include a Location header identifying the location of the newly created resource."

However this request returns a 201 status code and the location is not being rewritten.

**Comment by jrnail23** [comment]
related: #845
